chore: tidy tests following move from aws.ami --> aws.account

ajkerrigan opened the pull request:
## :dart: What needed to be done and why?

Clean up remnants of previous `aws.ami` work

*Quick summary of the issue*

## :new: What is changed by this PR?

- Remove flight recordings that are no longer referenced
- Update test comments

## :clipboard: Code Review Cheatsheet

<details>

<summary>What the reviewer should check</summary>

| Check  | Description |
| ------------- | ------------- |
| :truck: **Diff size** | Is the PR small and focused, or should it be broken into smaller PRs?
| :test_tube: **Unit tests** | Are new features covered with appropriate unit tests?
| :lab_coat: **Acceptance Criteria met** | Are the Acceptance Criteria listed in the issue met?
| :monocle_face: **Code clarity** | Is the code easy to understand? Are variable and function names meaningful?
| :jigsaw: **Code organization** | Are files, modules, and functions structured logically?
| :carpentry_saw: **Conciseness** | Is the code free of unnecessary complexity or redundant code?
| :speech_balloon: **Comments & documentation** | Are code comments explaining *why* and not *how*? Is the documentation up-to-date?
| :abacus: **Code consistency** | Does the code follow existing patterns and conventions in the project?
| :biohazard: **Function length** | Are functions too long? Should they be broken into smaller, more focused functions?
| :radioactive: **Class design** | Are classes well-structured and not overly large or doing too much?
| :paw_prints: **Logging and debugging statements** | Are print/debug statements removed or replaced with proper logging?

</details>

michaelsixfeetup commented on Makefile:
```diff
@@ -51,6 +51,10 @@ test-functional:
 
+test-functional-azure:
+# note this will provision real resources in Azure's public cloud environment
+	C7N_FUNCTIONAL=yes pytest tools/c7n_azure/tests_azure/tests_resources/test_entraid.py -k terraform -m functional $(ARGS)
```
@ratmav I didn't see it in the `test-function` block above for AWS, so I didn't add it here in an effort to keep the convention the same.

michaelsixfeetup commented on tools/c7n_azure/c7n_azure/graph_utils.py:
```diff
@@ -0,0 +1,154 @@
+    if normalized_endpoint in GRAPH_ENDPOINT_PERMISSIONS:
+        return GRAPH_ENDPOINT_PERMISSIONS[normalized_endpoint]
+    
+    # Check for pattern matches
```
This was the goal within `graph_utils.py` - the policies should not be able to access anything aside from the following:

```
GRAPH_ENDPOINT_PERMISSIONS = {}
```

michaelsixfeetup commented on tools/c7n_azure/c7n_azure/resources/entraid.py:
```diff
@@ -0,0 +1,91 @@
+**Common EntraID Policy Examples**
+
+Find guest users:
+.. code-block:: yaml
```
I prefer having the docs close by, but we can remove them if you'd prefer.

michaelsixfeetup commented on tools/c7n_azure/examples/admin-group-members.yaml:
```diff
@@ -0,0 +1,37 @@
+policies:
```
These are for us for testing, but I can remove them if you'd like.

michaelsixfeetup commented on tools/c7n_azure/tests_azure/conftest.py:
```diff
@@ -0,0 +1,72 @@
+class TerraformAzureRewriteHooks:
+    """Local pytest plugin for Azure terraform tests
+    
+    Work around to allow for dynamic registration of hooks based on plugin availability
```
The conftest.py file is pytest's configuration file that sets up the testing environment. You can see more info here: https://github.com/sixfeetup/stacklet-entraid/issues/44#issuecomment-3128640687

michaelsixfeetup commented:
Spoke with Chris on August 1st during EntraID standup - resolved all issues with him on the call (except the linting).

ajkerrigan committed:
tidy tests following move from aws.ami --> aws.account

- Remove flight recordings that are no longer referenced
- Update test comments

toastdriven commented:
Oops! Thank you very much for the fixes. Merged!
